zcash_primitives/src/note_encryption.rs

```diff
@@ -438,7 +481,7 @@ pub fn try_sapling_compact_note_decryption(
     plaintext.copy_from_slice(&enc_ciphertext);
     ChaCha20Ietf::xor(key.as_bytes(), &[0u8; 12], 1, &mut plaintext);
 
-    parse_note_plaintext_without_memo(ivk, cmu, &plaintext)
+    parse_note_plaintext_without_memo::<P>(height, ivk, cmu, &plaintext)
```

Comment:
The ZIP 212 check in `try_sapling_note_decryption` needs to be enforced here as well. Move it inside `parse_note_plaintext_without_memo` (just after the `cmu` check there).